### PR TITLE
mes-3827-Typescript and schema generation, new intersection type

### PR DESCRIPTION
**Pull request by amobarak**
# Description and relevant Jira numbers

Schema combination and typescript generation, new script and intersection type.
Categories B/BE/C are all refactored to use the common schema, all schemas outputted should pass validation of the original index.json before the common refactoring.

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: mes-test-schema/categories/BE/partial.json [hunk 1]
```diff
@@ -0,0 +1,229 @@
+{
+  "title": "Partial Standard Trailer Test (CAT B+E) Schema",
```
Comment (szabolcs-keresztes) [inline mes-test-schema/categories/BE/partial.json]: Can we simplify the name? As we mentioned on _appy-tappers_ channel please

File: mes-test-schema/categories/B/partial.json [hunk 2]
```diff
@@ -0,0 +1,251 @@
+{
+  "title": "Partial Standard Car Test (CAT B) Schema",
```
Comment (szabolcs-keresztes) [inline mes-test-schema/categories/B/partial.json]: Simplify the name as we mentioned on _appy-tappers_ channel please

**Comment by Cavalier07** [comment]
Can we remove all the .DS_Store files please - they should be on a .gitignore if they aren't already

File: mes-test-schema/package.json [hunk 3]
```diff
@@ -21,8 +20,13 @@
     "url": "https://github.com/dvsa/mes-api-definitions/issues"
   },
   "homepage": "https://github.com/dvsa/mes-api-definitions#readme",
-  "dependencies": {},
+  "dependencies": {
+    "deepmerge": "^4.1.1"
```
Comment (Cavalier07) [inline mes-test-schema/package.json]: should be a dev dependency

**Comment by ammarhaiderbjss** [comment]
@amobarak Could we please add some notes to the readme detailing the new process for generating schemas?

**Pull request merged**